app/commands/payments/subscription/create.rb

```diff
@@ -2,22 +2,25 @@
 # and creates a record of it in our database. The actual
 # creation of the subscription within Stripe happens through
 # "payment intents".
-class Donations::Subscription::Create
+class Payments::Subscription::Create
   include Mandate
 
-  initialize_with :user, :provider, :external_id, :amount_in_cents
+  initialize_with :user, :provider, :product, :interval, :external_id, :amount_in_cents
 
   def call
-    Donations::Subscription.create!(
+    Payments::Subscription.create!(
       user:,
       provider:,
+      product:,
+      interval:,
       external_id:,
       amount_in_cents:,
       status: :active
     ).tap do
-      User::UpdateActiveDonationSubscription.(user)
+      User::UpdateActiveDonationSubscription.(user) if product == :donation
+      User::Premium::Update.(user) if product == :premium
```

Comment:
Maybe a case on `product` or something here, just as they're mutually exclusive? I'm not fussed though.